support/azure/azure-kubernetes/availability-performance/node-auto-repair-errors.md

```diff
@@ -0,0 +1,40 @@
+---
+title: Troubleshoot common node auto-repair errors
+description: Troubleshoot scenarios where node auto-repair returns an error code when trying to repair your NotReady node.
+ms.date: 10/01/2024
+ms.reviewer: 
+ms.service: azure-kubernetes-service
+#Customer intent: As an Azure Kubernetes user, I want to make sure the automatic repair actions from AKS node auto-repair do not cause any impacts on my applications or cluster health.
+ms.custom: sap:Node/node pool availability and performance
+---
+# Troubleshoot common node auto-repair errors
+
+When AKS detects a node which reports the NotReady status for more than 5 minutes, we will attempt to automatically repair your node. Learn more about the node auto-repair process [here](https://learn.microsoft.com/en-us/azure/aks/node-auto-repair).
+
+During this process, AKS will initiate reboot, reimage, and redeploy actions on your unhealthy node. These repair actions may be unsuccessful due to an underlying cause, resulting in an error code. This article will discuss common errors with their potential causes and next steps, as well as best practices for monitoring node auto-repair.
+
+## Prerequisites
+To determine what type of node auto-repair error has occured, you will need to look for the following Kubernetes event: 
+"Node auto-repair [reboot/reimage/redeploy] action failed due to an operation failure: [error code]."
+
+## Common error codes
+The table below contains the most common node auto-repair errors.
+
+| Error code | Potential causes | Next steps |
+|---|---|---|
+| ClientSecretCredential authentication failed |  |  |
```

Comment:
as mentioned on teams, I think we can remove this one and add some other top errors. Since this is an issue with misclassifying MSI clusters which there is a fix rolling out for.